validation/src/lib.rs

```diff
@@ -348,7 +346,8 @@ impl<C, N, P> ParachainValidation<C, N, P> where
 
 		debug!(target: "validation", "Active parachains: {:?}", active_parachains);
 
-		let table = Arc::new(SharedTable::new(group_info, sign_with.clone(), parent_hash, self.extrinsic_store.clone()));
+		let index_mapping = authorities.iter().enumerate().map(|(i, k)| (i as ValidatorIndex, k.clone())).collect();
```

Comment:
Is this correct?

Reply:
yes, I believe it is. It requires that everybody have the same `authorities` vector, but since it is coming from a runtime API it should be. Please add a test